ebcl/common/files.py

```diff
@@ -210,6 +211,12 @@ def copy_file(
                     self._run_cmd(f'chown {uid} {target}', environment)
                 if gid:
                     self._run_cmd(f'chown :{gid} {target}', environment)
+
+                if not mode and not move:
+                    # Take over mode from source file.
+                    mode = oct(os.stat(file).st_mode)
+                    mode = mode[-3:]
```

Comment:
```suggestion
                    mode = mode[-4:]
```
This keeps the suid, sgid and sticky bit as well